Sources/SocksCore/Socket.swift

```diff
@@ -42,7 +42,10 @@ extension Socket {
         if config.reuseAddress {
             try setOption(descriptor: descriptor, level: SOL_SOCKET, name: SO_REUSEADDR, value: 1)
         }
-        
+
+        // prevents SIGPIPE from killing process
+        try setOption(descriptor: descriptor, level: SOL_SOCKET, name: SO_NOSIGPIPE, value: 1)
```

Comment:
Actually, do we even need this if we're handling it down in `send`? Can you please test out if `EPIPE` is returned from `send` even if this is not set on OS X? (I'd prefer to just do it directly in send if possible)